src/core/maprenderer/qgsmaprendererjob.cpp

```diff
@@ -374,6 +374,8 @@ QImage *QgsMapRendererJob::allocateImage( QString layerId )
   QImage *image = new QImage( mSettings.deviceOutputSize(),
                               mSettings.outputImageFormat() );
   image->setDevicePixelRatio( static_cast<qreal>( mSettings.devicePixelRatio() ) );
+  image->setDotsPerMeterX( 1000 * mSettings.outputDpi() / 25.4 );
```

Comment:
Doesn't this also need
 
```
    1000 * mSettings.outputDpi() / 25.4 * mSettings.devicePixelRatio()
```

? (as per qgsmaprenderersequentialjob.cpp:32)

Reply:
Yes, right!